<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from environment.urdf.xacro         | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="manipuation_environment">
  <!-- world -->
  <link name="world"/>
  <!-- hall -->
  <!--
  <xacro:include filename="$(find ros_industrial_support)/urdf/hall/hall.urdf.xacro"/>
  <xacro:hall_urdf hall_prefix="hall_" hall_parent="world">
    <origin xyz="-2.5 0 0.0" rpy="0 0 0"/>
  </xacro:hall_urdf>
  -->
  <joint name="assembly_station_joint" type="fixed">
    <origin rpy="0 0 1.5707963267948966" xyz="0.5 -0.5 0.0"/>
    <parent link="world"/>
    <child link="assembly_station_origin_pre"/>
  </joint>
  <link name="assembly_station_origin_pre"/>
  <joint name="assembly_station_joint_pre" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 0.0 0.03"/>
    <parent link="assembly_station_origin_pre"/>
    <child link="assembly_station_origin"/>
  </joint>
  <!-- Links -->
  <link name="assembly_station_origin">
    <visual>
      <geometry>
        <mesh filename="file:///home/student/ros2_industrial_ws/install/ros_industrial_support/share/ros_industrial_support/meshes/assembly_station/meshes/assembly_station.dae" scale="1.1 1.1 1.1"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/student/ros2_industrial_ws/install/ros_industrial_support/share/ros_industrial_support/meshes/assembly_station/meshes/assembly_station.dae" scale="1.1 1.1 1.1"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="1.0" ixy="0" ixz="0" iyy="1.0" iyz="0" izz="1.0"/>
      <mass value="100"/>
    </inertial>
  </link>
  <joint name="comp_mobile_joint" type="fixed">
    <origin rpy="0 0 0.7853981633974483" xyz="1.5 -0.5 0.00"/>
    <parent link="world"/>
    <child link="comp_mobile_origin"/>
  </joint>
  <!-- Links -->
  <!--
<link name="${computer_parent}"/>
-->
  <link name="comp_mobile_origin">
    <visual>
      <geometry>
        <mesh filename="file:///home/student/ros2_industrial_ws/install/ros_industrial_support/share/ros_industrial_support/meshes/computer_mobile/meshes/computer.dae" scale="0.01 0.01 0.01"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/student/ros2_industrial_ws/install/ros_industrial_support/share/ros_industrial_support/meshes/computer_mobile/meshes/computer.dae" scale="0.01 0.01 0.01"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="1.0" ixy="0" ixz="0" iyy="1.0" iyz="0" izz="1.0"/>
      <mass value="1.0"/>
    </inertial>
  </link>
  <gazebo>
    <static>true</static>
  </gazebo>
  <joint name="robot1__pedestal_joint" type="fixed">
    <!--
    <xacro:insert_block name="origin" />
    -->
    <parent link="world"/>
    <child link="robot1_pedestal_link"/>
  </joint>
  <link name="robot1_pedestal_link">
    <visual>
      <origin xyz="0 0 0.575"/>
      <geometry>
        <box size="0.3 0.3 1.15"/>
      </geometry>
      <material name="pedestal_color">
        <color rgba="0 0.1568627450980392 0.5254901960784314 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0.575"/>
      <geometry>
        <box size="0.3 0.3 1.15"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="500"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="robot1_pedestal_link">
    <material>Gazebo/Blue</material>
  </gazebo>
  <joint name="camera1_joint" type="fixed">
    <origin rpy="0 1.5707963267948966 0" xyz="0.5 -0.7 2.0"/>
    <parent link="world"/>
    <child link="camera1_origin"/>
  </joint>
  <link name="camera1_origin"/>
  <joint name="camera1_joint_post" type="fixed">
    <origin rpy="0 0 -1.5708" xyz="-0.02 0 -0.02"/>
    <parent link="camera1_origin"/>
    <child link="camera1_origin_post"/>
  </joint>
  <!-- Links -->
  <link name="camera1_origin_post">
    <visual>
      <geometry>
        <mesh filename="file:///home/student/ros2_industrial_ws/install/ros_industrial_support/share/ros_industrial_support/meshes/camera/visual/camera.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/student/ros2_industrial_ws/install/ros_industrial_support/share/ros_industrial_support/meshes/camera/collision/camera.dae"/>
      </geometry>
    </collision>
    <inertial>
      <inertia ixx="1.0" ixy="0" ixz="0" iyy="1.0" iyz="0" izz="1.0"/>
      <mass value="1.0"/>
    </inertial>
  </link>
  <gazebo>
    <static>true</static>
  </gazebo>
  <!-- Insert at the beginning of xarm_device_macro.xacro -->
  <material name="Black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="Red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="White">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <material name="Silver">
    <color rgba="0.753 0.753 0.753 1.0"/>
  </material>
  <joint name="world_joint" type="fixed">
    <parent link="base_link"/>
    <child link="link_base"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <ros2_control name="gazebo_ros2_control/GazeboSystem" type="system">
    <hardware>
      <plugin>gazebo_ros2_control/GazeboSystem</plugin>
      <!-- fake -->
      <!-- <plugin>fake_components/GenericSystem</plugin> -->
      <!-- gazebo -->
      <!-- <plugin>gazebo_ros2_control/GazeboSystem</plugin> -->
      <!-- real xarm -->
      <!-- <plugin>uf_robot_hardware/UFRobotSystemHardware</plugin> -->
    </hardware>
    <joint name="joint1">
      <command_interface name="position">
        <param name="min">-6.283185307179586</param>
        <param name="max">6.283185307179586</param>
      </command_interface>
      <command_interface name="velocity">
        <param name="min">-3.14</param>
        <param name="max">3.14</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <!-- <state_interface name="effort"/> -->
    </joint>
    <joint name="joint2">
      <command_interface name="position">
        <param name="min">-2.059</param>
        <param name="max">2.0944</param>
      </command_interface>
      <command_interface name="velocity">
        <param name="min">-3.14</param>
        <param name="max">3.14</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <!-- <state_interface name="effort"/> -->
    </joint>
    <joint name="joint3">
      <command_interface name="position">
        <param name="min">-3.927</param>
        <param name="max">0.19198</param>
      </command_interface>
      <command_interface name="velocity">
        <param name="min">-3.14</param>
        <param name="max">3.14</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <!-- <state_interface name="effort"/> -->
    </joint>
    <joint name="joint4">
      <command_interface name="position">
        <param name="min">-6.283185307179586</param>
        <param name="max">6.283185307179586</param>
      </command_interface>
      <command_interface name="velocity">
        <param name="min">-3.14</param>
        <param name="max">3.14</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <!-- <state_interface name="effort"/> -->
    </joint>
    <joint name="joint5">
      <command_interface name="position">
        <param name="min">-1.69297</param>
        <param name="max">3.141592653589793</param>
      </command_interface>
      <command_interface name="velocity">
        <param name="min">-3.14</param>
        <param name="max">3.14</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <!-- <state_interface name="effort"/> -->
    </joint>
    <joint name="joint6">
      <command_interface name="position">
        <param name="min">-6.283185307179586</param>
        <param name="max">6.283185307179586</param>
      </command_interface>
      <command_interface name="velocity">
        <param name="min">-3.14</param>
        <param name="max">3.14</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <!-- <state_interface name="effort"/> -->
    </joint>
  </ros2_control>
  <link name="link_base">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.09103"/>
      <mass value="2.7"/>
      <inertia ixx="0.00494875" ixy="-3.5E-06" ixz="1.25E-05" iyy="0.00494174" iyz="1.67E-06" izz="0.002219"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link_base.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="White"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link_base.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <link name="link1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.00022 0.02951 -0.0124"/>
      <mass value="2.3814"/>
      <inertia ixx="0.0058562" ixy="-1.79e-05" ixz="3.55e-06" iyy="0.0050316" iyz="-0.000888336" izz="0.003536652"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link1.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="White"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link1.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="joint1" type="revolute">
    <parent link="link_base"/>
    <child link="link1"/>
    <origin rpy="0 0 0" xyz="0 0 0.267"/>
    <axis xyz="0 0 1"/>
    <limit effort="50.0" lower="-6.283185307179586" upper="6.283185307179586" velocity="3.14"/>
    <dynamics damping="1.0" friction="1.0"/>
  </joint>
  <link name="link2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.03881 -0.22783 0.03496"/>
      <mass value="2.2675"/>
      <inertia ixx="0.028315776" ixy="0.005" ixz="0.00066546" iyy="0.0058" iyz="-0.0045741" izz="0.0273447"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link2.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="White"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link2.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="joint2" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin rpy="-1.5708 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="50.0" lower="-2.059" upper="2.0944" velocity="3.14"/>
    <dynamics damping="1.0" friction="1.0"/>
  </joint>
  <link name="link3">
    <inertial>
      <origin rpy="0 0 0" xyz="0.07041 0.11631 0.0107"/>
      <mass value="1.875"/>
      <inertia ixx="0.0063483" ixy="-0.0015397" ixz="0.00096858" iyy="0.00379758" iyz="0.00186567" izz="0.00595768"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link3.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="White"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link3.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="joint3" type="revolute">
    <parent link="link2"/>
    <child link="link3"/>
    <origin rpy="0 0 0" xyz="0.0535 -0.2845 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="32.0" lower="-3.927" upper="0.19198" velocity="3.14"/>
    <dynamics damping="1.0" friction="1.0"/>
  </joint>
  <link name="link4">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00018 0.01798 -0.02291"/>
      <mass value="1.3192"/>
      <inertia ixx="0.004896" ixy="-6.925e-06" ixz="-1.418e-05" iyy="0.00445694" iyz="-0.00023186" izz="0.00134332"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link4.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="White"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link4.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="joint4" type="revolute">
    <parent link="link3"/>
    <child link="link4"/>
    <origin rpy="-1.5708 0 0" xyz="0.0775 0.3425 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="32.0" lower="-6.283185307179586" upper="6.283185307179586" velocity="3.14"/>
    <dynamics damping="1.0" friction="1.0"/>
  </joint>
  <link name="link5">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0651 0.03096 0.00315"/>
      <mass value="1.33854"/>
      <inertia ixx="0.00146378" ixy="-0.000450624" ixz="0.000284306" iyy="0.00184192" iyz="0.000130866" izz="0.002333524"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link5.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="White"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link5.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="joint5" type="revolute">
    <parent link="link4"/>
    <child link="link5"/>
    <origin rpy="1.5708 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="32.0" lower="-1.69297" upper="3.141592653589793" velocity="3.14"/>
    <dynamics damping="1.0" friction="1.0"/>
  </joint>
  <link name="link6">
    <inertial>
      <origin rpy="0 0 0" xyz="0 -0.00677 -0.01098"/>
      <mass value="0.17"/>
      <inertia ixx="9.3e-05" ixy="-0.0" ixz="-0.0" iyy="5.87e-05" iyz="-3.6e-06" izz="0.000132"/>
    </inertial>
    <visual>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/xarm6/visual/link6.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="Silver"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/student/xarm_ws/install/xarm_description/share/xarm_description/meshes/end_tool/collision/end_tool.stl"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
  </link>
  <joint name="joint6" type="revolute">
    <parent link="link5"/>
    <child link="link6"/>
    <origin rpy="-1.5708 0 0" xyz="0.076 0.097 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="20.0" lower="-6.283185307179586" upper="6.283185307179586" velocity="3.14"/>
    <dynamics damping="1.0" friction="1.0"/>
  </joint>
  <!-- <link name="${prefix}link_eef" />
    <joint name="${prefix}joint_eef" type="fixed">
      <origin xyz="0 0 0" rpy="0 0 0" />
      <parent link="${prefix}link6" />
      <child link="${prefix}link_eef" />
    </joint> -->
  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran3">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran4">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint4">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor4">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran5">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint5">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor5">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran6">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint6">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor6">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>100</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="link_base">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="link1">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="link2">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="link3">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="link4">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="link5">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="link6">
    <selfCollide>true</selfCollide>
  </gazebo>
  <link name="link_eef"/>
  <joint name="joint_eef" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link6"/>
    <child link="link_eef"/>
  </joint>
  <gazebo>
    <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
      <robot_sim_type>gazebo_ros2_control/GazeboSystem</robot_sim_type>
      <parameters>/home/student/xarm_ws/install/xarm_controller/share/xarm_controller/config/xarm6_controllers.yaml</parameters>
    </plugin>
  </gazebo>
  <!-- Insert at the end of xarm_device_macro.xacro -->
  <link name="base_link"/>
  <link name="tool_link"/>
  <joint name="eff_to_tool" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="link_eef"/>
    <child link="tool_link"/>
  </joint>
  <!-- dit werkt ook niet -->
  <gazebo reference="base_link">
    <material>Gazebo/Black</material>
  </gazebo>
  <gazebo reference="tool_link">
    <material>Gazebo/Black</material>
  </gazebo>
  <gazebo reference="link_eef">
    <material>Gazebo/Black</material>
  </gazebo>
  <!-- link list -->
  <link name="vacuum_gripper1_base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/student/ros2_industrial_ws/install/ros_industrial_support/share/ros_industrial_support/meshes/vacuum_gripper/visual/vacuum_gripper.dae"/>
      </geometry>
      <material name="red">
        <color rgba="1 0 0 1"/>
      </material>
    </visual>
    <collision name="vacuum_gripper1_base_link_collision">
      <!--
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <geometry>
          <mesh filename="${mesh_path}/meshes/vacuum_gripper/collision/vacuum_gripper.dae"/>
        </geometry>
        <material name="red"/>
        -->
      <geometry>
        <!-- <mesh filename="file://$(find ariac_description)/meshes/vacuum_gripper/vacuum_gripper.stl" scale="1 1 1"/> -->
        <cylinder length="0.01" radius="0.05"/>
      </geometry>
    </collision>
    <!--
      <inertial>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <inertia ixx="1.0" ixy="0" ixz="0"
                 iyy="1.0" iyz="0"
                 izz="1.0" />
        <mass value="1.0" />
      </inertial>
      -->
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="0.0000063" ixy="0.0" ixz="0.0" iyy="0.0000063" iyz="0.0" izz="0.000025"/>
    </inertial>
  </link>
  <link name="vacuum_gripper1_suction_cup">
    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin xyz="0 0 0"/>
      <geometry>
        <box size="0.001 0.001 0.001"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
      <mass value="0.001"/>
    </inertial>
  </link>
  <!-- joint list -->
  <joint name="vacuum_gripper1_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.08"/>
    <parent link="vacuum_gripper1_base_link"/>
    <child link="vacuum_gripper1_suction_cup"/>
  </joint>
  <gazebo reference="vacuum_gripper1_suction_cup">
    <sensor name="bumper" type="contact">
      <update_rate>10</update_rate>
      <contact>
        <collision>vacuum_gripper1__gripper_collision_collision</collision>
      </contact>
    </sensor>
  </gazebo>
  <gazebo reference="vacuum_gripper1_base_link">
    <material>Gazebo/Green</material>
  </gazebo>
  <gazebo reference="vacuum_gripper1_suction_cup">
    <material>Gazebo/Black</material>
  </gazebo>
  <gazebo>
    <!--<plugin name="vacuum_gripper" filename="libgazebo_ros_vacuum_gripper.so">-->
    <plugin filename="libMyVacuumGripperPlugin.so" name="vacuum_gripper">
      <ros>
        <!--<namespace>/gazebo_plugins</namespace>-->
        <namespace>/ros_industrial_actuators</namespace>
        <remapping>switch:=custom_switch</remapping>
        <remapping>grasping:=custom_grasping</remapping>
      </ros>
      <link_name>vacuum_gripper1_suction_cup</link_name>
      <!--<link_name>vacuum_gripper1_suction_cup</link_name>-->
      <!-- Max distance to attract entities -->
      <max_distance>10.0</max_distance>
    </plugin>
  </gazebo>
  <!-- link list -->
  <link name="bin_1_base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/student/ros2_industrial_ws/install/ros_industrial_support/share/ros_industrial_support/meshes/bin/visual/bin.dae"/>
        <scale>0.6 0.6 0.6</scale>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/student/ros2_industrial_ws/install/ros_industrial_support/share/ros_industrial_support/meshes/bin/collision/bin_bottom.dae"/>
        <scale>0.6 0.6 0.6</scale>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="file:///home/student/ros2_industrial_ws/install/ros_industrial_support/share/ros_industrial_support/meshes/bin/collision/bin_sides.dae"/>
        <scale>0.6 0.6 0.6</scale>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="1.0" ixy="0" ixz="0" iyy="1.0" iyz="0" izz="1.0"/>
      <mass value="100"/>
    </inertial>
  </link>
  <link name="bin_1_drop_point">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="1.0" ixy="0" ixz="0" iyy="1.0" iyz="0" izz="1.0"/>
      <mass value="1.0"/>
    </inertial>
  </link>
  <!-- joint list -->
  <joint name="bin_1_joint_1" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 1.0"/>
    <parent link="bin_1_base_link"/>
    <child link="bin_1_drop_point"/>
  </joint>
  <!-- bin to world. -->
  <joint name="bin_1_joint" type="fixed">
    <parent link="world"/>
    <child link="bin_1_base_link"/>
    <origin rpy="0 0 0" xyz="-0.5 0.5 0.0"/>
  </joint>
  <!-- robot1 to pedestal. -->
  <joint name="robot1_to_pedestal" type="fixed">
    <origin rpy="0 0 0.0" xyz="0 0 1.15"/>
    <parent link="robot1_pedestal_link"/>
    <child link="base_link"/>
  </joint>
  <!-- gripper1 to robot1. -->
  <joint name="gripper1_to_robot1" type="fixed">
    <parent link="tool_link"/>
    <child link="vacuum_gripper1_base_link"/>
  </joint>
</robot>
